### ISSUE TITLE
Conflicts with pyarrow 4.0?

### ISSUE THREAD

**Issue by ldacey** [blue-yonder/turbodbc#307]
```
name: test_pyarrow
channels:
  - conda-forge
dependencies:
  - python=3.8
  - pyarrow=4
  - turbodbc
```

I was trying to update pyarrow and I ran into conflicts with this minimal environment. Is there a way for me to force the installation, or does one of the libraries need to be updated?

**Comment by xhochy** [comment]
We need to change https://github.com/blue-yonder/turbodbc/blob/2deec5e2e42b7842b4a3c0c9587280dcc922d97b/setup.py#L218 and then rebase https://github.com/conda-forge/turbodbc-feedstock/pull/51/files on top of this.

**Comment by alippai** [comment]
Filed https://github.com/blue-yonder/turbodbc/pull/308, changed the numpy and python requirements based on NEP 29 as well (and updated the docs)

**Comment by xhochy** [comment]
Released as 4.2.0

**Comment by alippai** [comment]
Thanks @xhochy!
